Assets/CodeBase/Entities/player/Player.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Player : MonoBehaviour
+{
+    [SerializeField]
+    float moveSpeed;
+
+    [SerializeField]
+    float jumpSpeed;
+
+    PlayerInputProfile inputProfile;
+    Rigidbody2D playerRigidBody;
+
+    private bool _isGrounded;
+    // Start is called before the first frame update
+    void Start()
+    {
+        inputProfile = new PlayerInputProfile();
+        playerRigidBody = GetComponent<Rigidbody2D>();
+
+        _isGrounded = false;
+
+        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.moveUp, moveUp);
+        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.moveDown, moveDown);
+        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.moveLeft, moveLeft);
+        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.moveRight, moveRight);
+        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.jump, jump);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        inputProfile.checkInput();
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        Debug.Log("Fuck");
```

Comment:
Dont forget to clean logs we dont want the grader to see these LOL

Reply:
hhahaha